intro-HPC/ch_MATLAB.tex

```diff
@@ -0,0 +1,117 @@
+\chapter{MATLAB}
+\label{ch:matlab}
+
+\section{Why is the MATLAB compiler required?}
+
+The main reason behind this alternative way of using MATLAB is licensing: only
+a limited number of MATLAB sessions can be active at the same time. However, once
+the MATLAB program is compiled using the MATLAB compiler, the resulting stand-alone
+executable can be run without needing to contact the license server.
+A license is required for the MATLAB Compiler, see \url{https://nl.mathworks.com/help/compiler/index.html}.
+
+\ifgent
+Only a limited amount of MATLAB sessions can be active at the same time because
+there are only a limited amount of MATLAB research licences available on
+the \university MATLAB license server. Since there are a lot of compute nodes,
+if each node would need a license, the licenses would quickly run out.
+\fi
+
+\section{How to compile MATLAB code}
+
+Compiling MATLAB code can only be done from the login nodes, because only login
+nodes can access the server, workernodes on clusters cannot.
+
+To access the MATLAB compiler, the MATLAB module should be loaded first. Make sure
+you are using the same MATLAB version to compile and to run the compiled MATLAB
+program.
+
+\begin{prompt}
+%\shellcmd{module avail MATLAB}%
+----------------------/apps/gent/CO7/sandybridge/modules/all ----------------------
```

Comment:
Since this will be different for each site, should I make an issue?

Reply:
Default paths for each sites, you could use these in the macro:

* KUL: `/apps/leuven/thinking/2014a/modules/all`
* VUB: `/apps/brussel/CO7/magnycours-ib/modules/2017a/all`
* UA: `/apps/antwerpen/modules/noarch/all`

Or maybe we should use a generic one, something like `/apps/<SITE>/\defaultcluster/modules/all` (where `<SITE>` is replaced).
That's probably a better options, since i) this will change over time, ii) it may depend on the software (see the `noarch` for UA).